src/main/java/step2/domain/Lotto.java

```diff
@@ -0,0 +1,41 @@
+package step2.domain;
+
+import java.util.List;
+
+public class Lotto {
+
+    private final List<LottoNumber> numbers;
+    private static final int LOTTO_SIZE = 6;
+    public Lotto(List<LottoNumber> numbers) {
+        inputCheck(numbers);
+        this.numbers = numbers;
+    }
+
+    private void inputCheck(List<LottoNumber> numbers) {
+        if(numbers.size() != LOTTO_SIZE){
+            throw new IllegalArgumentException("로또 번호는 6개만 가능합니다.");
+        }
+
+        if(numbers.stream().distinct().count() != LOTTO_SIZE){
+            throw new IllegalArgumentException("로또 번호는 중복될 수 없습니다.");
+        }
+    }
+
+    public List<LottoNumber> lottoNumbers() {
+        return this.numbers;
+    }
+
+    public int matchCount(Lotto winningNumbersLotto) {
```

Comment:
```suggestion
    public LottoWinningInfo matchCount(Lotto winningNumbersLotto) {
```

지금도 좋지만
여기서 enum을 반환해도 될 것 같지 않나요?

Reply:
이러면 WinningLotto의 복잡한 로직이 많이 개선될 것 같아요